### ISSUE TITLE
Unexpected box result of cls and conf after export onnx

### ISSUE THREAD

**Issue by chilin0525** [ultralytics/ultralytics#16968]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

After I expose custom model weight after fine-tune on my dataset from `pt` to `onnx`:

```python
from ultralytics import YOLO
model = YOLO("best.pt")
model.export(format="onnx", imgsz=640, int8=False, simplify=True, dynamic=True)
onnx_model = YOLO("best.onnx")
```

I got log from which is successfully to export to onnx,

```
Ultralytics 8.3.14 🚀 Python-3.9.20 torch-2.4.1+cu121 CPU (Intel Xeon Gold 6444Y)
YOLOv8m-seg summary (fused): 245 layers, 27,243,228 parameters, 0 gradients, 110.1 GFLOPs

PyTorch: starting from 'best.pt' with input shape (1, 3, 640, 640) BCHW and output shape(s) ((1, 72, 8400), (1, 32, 160, 160)) (52.3 MB)

ONNX: starting export with onnx 1.17.0 opset 19...
ONNX: slimming with onnxslim 0.1.34...
ONNX: export success ✅ 15.4s, saved as 'best.onnx' (104.1 MB)

Export complete (16.2s)
Results saved to /home/jovyan/workspace/XXXX
Predict:         yolo predict task=segment model=best.onnx imgsz=640  
Validate:        yolo val task=segment model=best.onnx imgsz=640 data=/mnt/data/XXXX/dataset.yaml  
Visualize:       https://netron.app/
WARNING ⚠️ Unable to automatically guess model task, assuming 'task=detect'. Explicitly define task for your model, i.e. 'task=detect', 'segment', 'classify','pose' or 'obb'.
```

and also get correct class name from `model.names` and `onnx_model.names` (hide some information of name):
```
{0: 'xxx',
 1: 'xxx',
 2: 'xxx',
 ...
 34: 'xxx',
 35: 'xxx'}
```

But when using onnx runtime to do inference, I got unpected `cls` result and also incorrect `conf` which not normalized from 0 to 1.

```python
results = onnx_model.predict(
    'xxx.jpg',
    verbose=False,
)
```

sample from result:
```python
print(results[0].boxes[1])

ultralytics.engine.results.Boxes object with attributes:

cls: tensor([56.])   # <--  should not more than 35
conf: tensor([2.8479])  # <--- should be in range 0 to 1
data: tensor([[  1.2141, 458.9106,  19.2868, 563.5306,   2.8479,  56.0000]])
id: None
is_track: False
orig_shape: (680, 841)
shape: torch.Size([1, 6])
xywh: tensor([[ 10.2504, 511.2206,  18.0727, 104.6201]])
xywhn: tensor([[0.0122, 0.7518, 0.0215, 0.1539]])
xyxy: tensor([[  1.2141, 458.9106,  19.2868, 563.5306]])
xyxyn: tensor([[0.0014, 0.6749, 0.0229, 0.8287]])
```

`onnx` related package version list:
```
onnx                     1.17.0
onnxruntime              1.19.2
onnxslim                 0.1.34
```

Does something wrong in my code? 🤔

**Comment by chilin0525** [comment]
For someone interesting, the pretrained model work well after export to onnx using same code and same environment.
![image](https://github.com/user-attachments/assets/919c32f2-3830-4b28-a2e1-a09cdcf7c6b3)

**Comment by Y-T-G** [comment]
You have to pass `task="segment"` while loading the model. It should have shown a warning about that too.

**Comment by Y-T-G** [comment]
Did you use `model = YOLO("best.onnx", task="segment")`?

**Comment by chilin0525** [comment]
@Y-T-G  yes

**Comment by Y-T-G** [comment]
What model is this?

**Comment by Y-T-G** [comment]
Did you check whether `best.pt` works?

**Comment by chilin0525** [comment]
The model is YOLOv8. If using `best.pt` directly, the model works well, the following is result from `best.pt`:

```python
ultralytics.engine.results.Boxes object with attributes:

cls: tensor([3.])
conf: tensor([0.8287])
data: tensor([[ 37.6540, 111.2483, 156.2000, 381.9095,   0.8287,   3.0000]])
id: None
is_track: False
orig_shape: (680, 841)
shape: torch.Size([1, 6])
xywh: tensor([[ 96.9270, 246.5789, 118.5460, 270.6613]])
xywhn: tensor([[0.1153, 0.3626, 0.1410, 0.3980]])
xyxy: tensor([[ 37.6540, 111.2483, 156.2000, 381.9095]])
xyxyn: tensor([[0.0448, 0.1636, 0.1857, 0.5616]])
```

**Comment by Y-T-G** [comment]
Which version of `ultralytics` did you use for training?

**Comment by Y-T-G** [comment]
Can you try exporting `last.pt`?

**Comment by chilin0525** [comment]
* `8.0.176`
  * `best.pt` works
  * `best.onnx` not works
* `8.3.14`
  * `best.pt` works
  * `best.onnx` not works

**Comment by chilin0525** [comment]
Using `last.pt` still same issue (undefined `cls` and `conf` greater than 1)

**Comment by Y-T-G** [comment]
What if you export to `torchscript`?

**Comment by chilin0525** [comment]
@Y-T-G  Oh I sorry, I forgot add `task` when load from onnx... now is work. Really thank for your support!
